### PR TITLE
ci: Add mypy

### PR DESCRIPTION
**Pull request by ericf-firebolt**
Resolves #

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue.
- [x] This PR includes tests, or tests are not required/relevant for this PR.
- [x] I have updated `CHANGELOG.md` and added information about my change.
- [x] I have removed any print or log calls that were added for debugging.
- [x] If this PR requires a new PyPI release I have bumped the version number.
- [x] I have verified that this PR contains only code changes relevant to this PR.
- [x] If further integration tests are now passing I've edited firebolt.dbtspec to account for this.
- [x] I have pulled/merged from the main branch if there are merge conflicts.
- [x] After pulling/merging main I have run pytest on the included or updated firebolt.dbtspec.

### CODE REVIEW

File: dbt/adapters/firebolt/connections.py [hunk 1]
```diff
@@ -114,19 +115,21 @@ def get_response(cls, cursor) -> AdapterResponse:
...
+    def commit(self) -> None:
         """
         Passing `SQLConnectionManager.begin()` because
         Firebolt does not yet support transactions.
         """
 
-    def cancel(self, connection):
+    def cancel(
+        self, connection: SQLConnectionManager
+    ) -> dbt.exceptions.NotImplementedException:
```
Comment (ptiurin) [inline dbt/adapters/firebolt/connections.py:132]: Does mypy allow this? My issue here is that we're not returning the Exception object, we're raising it. So technically the return type is -> None.

File: setup.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 config = read_configuration('setup.cfg')
 version = config['metadata']['version']
-dbt_version = version.rsplit('.', 1)[0]
-install_requires = [f'dbt-core~={dbt_version}', config['options']['install_requires']]
+install_requires = [config['options']['install_requires']]
```
Comment (ptiurin) [inline setup.py:7]: Is this supposed to be a part of this PR?

**Comment by ptiurin** [comment]
I think `dbt/include/py.typed` is redundant as in `setup.cfg` we're setting mypy to analyse `dbt/adapters` directory only.

File: setup.cfg [hunk 3]
```diff
@@ -40,12 +41,48 @@ dev =
...
+show_error_codes = True
+files = dbt/adapters
+namespace_packages = True
+explicit_package_bases = True
+
+[mypy-agate]
+ignore_missing_imports = True
+
+[mypy-dbt]
+ignore_missing_imports = True
```
Comment (ptiurin) [inline setup.cfg:66]: would this not ignore all the `dbt.*` imports as well?

File: setup.py (hunk 2 above)
Comment (ericf-firebolt) [reply setup.py:7]: No, that somehow got copied over from a different PR. Fixed.

File: dbt/adapters/firebolt/connections.py (hunk 1 above)
Comment (ericf-firebolt) [reply dbt/adapters/firebolt/connections.py:132]: Huh. Thought I'd already fixed that.

**Comment by ericf-firebolt** [comment]
> I think `dbt/include/py.typed` is redundant as in `setup.cfg` we're setting mypy to analyse `dbt/adapters` directory only.

Okay, that's weird. yesterday I couldn't get it to work without it. I deleted both `py.typed` files and all now seems good.

**Review by ptiurin — APPROVED** [review]

**Pull request merged**